src/decoder/mod.rs

```diff
@@ -167,6 +176,45 @@ mod tests {
         assert_eq!(data, output);
     }
 
+    #[tokio::test]
+    async fn test_async_gzip_decode() {
+        let data: Vec<u8> = b"This is some text!".to_vec();
+
+        let mut encoder = GzEncoder::new(Vec::new(), flate2::Compression::default());
+        encoder.write_all(&data).unwrap();
```

Comment:
Not related to this PR. Shall we use `expect()` rather than `unwrap()` in this repo? I think it can tell more information when CI or test fails.

Reply:
This code is for prepare the test data, we can ensure it works before following tests.